modify tests for Hotel.sol to test for onlyOwner

mikec opened the pull request:
The tests for non-owner execution were using the `callHotel` function in `WTIndex.sol`, so they were actually testing that `msg.sender` exists in the `hotelsByManager` mapping. This commit changes the tests in `hotel.js` to execute functions directly on the `Hotel.sol` contract with a non-owner, in order to test that it's public functions are protected by the `onlyOwner` modifier.

This commit also adds tests to `WTIndex.js` that test the `callHotel` function directly.

kvbik commented on test/WTIndex.js:
```diff
@@ -2,6 +2,7 @@ const chai = require('chai').assert;
 const help = require('./helpers/index.js');
 
 const WTIndex = artifacts.require('./WTIndex.sol');
+const WTHotel = artifacts.require('Hotel.sol');
```
i like the separation of the tests, so shouldn't `WTIndex.js` do `callHotel` instead of importing it?

adklempner replied:
Needs to be imported to build the call data

kvbik commented:
hey @mikec thx a lot for this PR! let's wait for @adklempner comments, too. but it looks like a good addition!

kvbik commented on test/hotel.js:
```diff
@@ -497,29 +480,27 @@ contract('Hotel', function(accounts) {
 
-    it('should fail if the Unit is not listed in the Hotels index of Units', async function(){
-      const unknownUnit = await Unit.new(wtHotel.address, typeNameHex, {from: hotelAccount});
+    it('should throw if not executed by owner', async function() {
```
looks to me like swapping the order of `should fail if the Unit is not listed in the Hotels index of Units` with `should throw if not executed by owner` should make the diff bit easier to read. because the latter is replacing `should fail if a non-owner calls the Unit` isn't it?

kvbik replied:
maybe i am mistaken on this one. it is just bit too much of change here :))) but we will navigate!

adklempner commented on test/hotel.js:
```diff
@@ -497,29 +480,27 @@ contract('Hotel', function(accounts) {
         assert(false);
       } catch(e) {
-        assert(help.isInvalidOpcodeEx(e));
+        assert(help.isInvalidOpcodeEx);
```
`help.isInvalidOpcodeEx` is a function and should take the error as an argument

adklempner commented on test/hotel.js:
```diff
@@ -497,29 +480,27 @@ contract('Hotel', function(accounts) {
+        await wtHotel.callUnit(unit.address, setPriceData, {from: nonOwnerAccount});
         assert(false);
       } catch(e) {
         assert(help.isInvalidOpcodeEx);
```
Replace with `help.isInvalidOpcodeEx(e)`

mikec commented:
@kvbik @adklempner thanks for the review! I changed the order of tests back to the original so the diff is a little easier to read now, and fixed the invalidOpcode asserts.

Let me know if you want me to squash the commits before this gets merged
